File: testapp/src/main/java/net/maxsmr/testapp/TestApplication.java
package net.maxsmr.testapp;

import android.app.Application;
import org.jetbrains.annotations.NotNull;

import net.maxsmr.commonutils.logger.BaseLogger;
import net.maxsmr.commonutils.logger.holder.BaseLoggerHolder;
import net.maxsmr.testapp.logger.ConfigureLog4J;
import net.maxsmr.testapp.logger.Slf4Logger;

import org.apache.log4j.Level;
import org.slf4j.LoggerFactory;

import java.io.File;

public class TestApplication extends Application {

    static {
        BaseLoggerHolder.initInstance(() -> new BaseLoggerHolder(true) {
            @Override
            protected BaseLogger createLogger(@NotNull String className) {
                if (ConfigureLog4J.getInstance().isInitialized()) {
                    return new Slf4Logger(LoggerFactory.getLogger(className));
                } else {
                    return null;
                }
            }
        });
    }

    @Override
    public void onCreate() {
        super.onCreate();
        ConfigureLog4J.getInstance().configure(Level.ALL, false, new File(getFilesDir(), "1.log").getAbsolutePath(), ConfigureLog4J.DEFAULT_MIN_FILE_SIZE, 1);
    }
}

File: testapp/src/main/kotlin/net/maxsmr/testapp/TestApplication.kt
package net.maxsmr.testapp

import android.app.Application
import net.maxsmr.commonutils.logger.BaseLogger
import net.maxsmr.commonutils.logger.holder.BaseLoggerHolder
import net.maxsmr.testapp.logger.ConfigureLog4J
import net.maxsmr.testapp.logger.Slf4Logger
import org.apache.log4j.Level
import org.slf4j.LoggerFactory
import java.io.File

class TestApplication: Application() {

    override fun onCreate() {
        super.onCreate()
        ConfigureLog4J.configure(Level.ALL,
                File (externalCacheDir ?: cacheDir, "test.log"))
    }

    companion object {

        init {
            // можно вызвать и после configure в onCreate
            // если обращение к getLogger здесь будет раньше configure -> падение
            BaseLoggerHolder.initInstance {
                object : BaseLoggerHolder() {

                    override fun createLogger(className: String): BaseLogger =
                            Slf4Logger(LoggerFactory.getLogger(className))
                }
            }
        }
    }
}